action-server/actions/daily_ci_enroll_form.py

```diff
@@ -270,14 +275,41 @@ def submit(
         tracker: Tracker,
         domain: Dict[Text, Any],
     ) -> List[Dict]:
-        if tracker.get_slot(DO_ENROLL_SLOT) is True:
+        if tracker.get_slot(DO_ENROLL_SLOT) is False:
+            return []
+
+        reminder = Reminder.create_from_slot_values(tracker.current_slot_values())
+
+        if _save_reminder(reminder):
             dispatcher.utter_message(template="utter_daily_ci_enroll__enroll_done_1")
             dispatcher.utter_message(template="utter_daily_ci_enroll__enroll_done_2")
             dispatcher.utter_message(template="utter_daily_ci_enroll__enroll_done_3")
+        else:
+            dispatcher.utter_message(template="utter_daily_ci_enroll__enroll_fail_1")
+            dispatcher.utter_message(template="utter_daily_ci_enroll__enroll_fail_2")
+            dispatcher.utter_message(template="utter_daily_ci_enroll__enroll_fail_3")
 
         return []
 
 
+def _save_reminder(reminder):
+    if reminder.phone_number[4:7] == "555":
```

Comment:
What is the allowed phone number format? At some point, we allowed both 10 and 11 digits. Not sure if that's still the case. We just have to make sure that the indexation above matches the allowed format, whether it's 10 or 11 digits.

Reply:
Indeed, I should assert the length. It should be exactly 11.